source/Calamari.Azure/ServicePrincipalAccount.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Calamari.Azure
 {
-    public class ServicePrincipalAccount
```

Comment:
There's no actual shared base project for Calamari Azure steps. This one is specifically for Kubernetes cloud target discovery and is only included in the base Calamari Project. I'd like to pull the ServicePrincipalAccount, AzureOidcAccount and ideally all accounts into Calamari.Common which is used in the majority of steps. I have not done this as a part of this PR, at least not yet any feedback or hesitations on making this change?

Reply:
An initial cut at this required bringing in the following packages to Calamari.Common. Alterntivly these could just be extension methods in the individual flavours that requite these (AzureWebApp & Calamari.Azure) It's likely worthwhile doing this but I'm holding off for feedback

`using Microsoft.Azure.Management.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
using Microsoft.Rest;`